pipelines/options.go

```diff
@@ -0,0 +1,309 @@
+	if c.logToStdout {
+		// there is no agent; all features off
+		return
+	}
+	resp, err := c.httpClient.Get(fmt.Sprintf("http://%s/info", c.agentAddr))
+	if err != nil {
+		log.Printf("ERROR: Loading features: %v", err)
+		return
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		// agent is older than 7.28.0, features not discoverable
+		return
+	}
+	defer resp.Body.Close()
+	type infoResponse struct {
+		Endpoints  []string `json:"endpoints"`
+		StatsdPort int      `json:"statsd_port"`
+	}
+	var info infoResponse
+	if err := json.NewDecoder(resp.Body).Decode(&info); err != nil {
+		log.Printf("ERROR: Decoding features: %v", err)
+		return
+	}
+	c.features.StatsdPort = info.StatsdPort
+	for _, endpoint := range info.Endpoints {
+		switch endpoint {
+		case "/v0.1/pipeline_stats":
+			c.features.PipelineStats = true
+			log.Printf("INFO: Enable pipeline stats.")
+		}
+	}
+}
+
+func statsTags(c *config) []string {
+	tags := []string{
+		"lang:go",
+		"lang_version:" + runtime.Version(),
+	}
+	if c.service != "" {
+		tags = append(tags, "service:"+c.service)
+	}
+	if c.env != "" {
+		tags = append(tags, "env:"+c.env)
+	}
+	if c.hostname != "" {
+		tags = append(tags, "host:"+c.hostname)
+	}
+	for k, v := range c.globalTags {
+		if vstr, ok := v.(string); ok {
+			tags = append(tags, k+":"+vstr)
+		}
+	}
+	return tags
+}
+
+// withNoopStats is used for testing to disable statsd client
+func withNoopStats() StartOption {
+	return func(c *config) {
+		c.statsd = &statsd.NoOpClient{}
+	}
+}
+
+// WithService sets the default service name for the program.
+func WithService(name string) StartOption {
+	return func(c *config) {
+		c.service = name
+	}
+}
+
+// WithAgentAddr sets the address where the agent is located. The default is
+// localhost:8126. It should contain both host and port.
+func WithAgentAddr(addr string) StartOption {
+	return func(c *config) {
+		c.agentAddr = addr
+	}
+}
+
+// WithEnv sets the environment to which all traces started by the tracer will be submitted.
+// The default value is the environment variable DD_ENV, if it is set.
+func WithEnv(env string) StartOption {
+	return func(c *config) {
+		c.env = env
+	}
+}
+
+// WithDogstatsdAddress specifies the address to connect to for sending metrics
+// to the Datadog Agent. If not set, it defaults to "localhost:8125" or to the
+// combination of the environment variables DD_AGENT_HOST and DD_DOGSTATSD_PORT.
+// This option is in effect when WithRuntimeMetrics is enabled.
+func WithDogstatsdAddress(addr string) StartOption {
+	return func(cfg *config) {
+		cfg.dogstatsdAddr = addr
+	}
+}
+
+// WithSite starts the pipeline processor with a given site to send data to.
+func WithSite(site string) StartOption {
+	return func(c *config) {
+		c.site = site
+	}
+}
+
+// WithAgentLess starts the pipeline processor in a mode where stats are sent directly to the datadog backend
+// instead of going through the agent.
+func WithAgentLess(apiKey string) StartOption {
```

Comment:
```suggestion
func WithAgentless(apiKey string) StartOption {
```
I believe this is one word, not two.